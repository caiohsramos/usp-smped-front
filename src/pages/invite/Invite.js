import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Container } from '../Container';
import { InviteController } from './InviteController';
import { bindActionCreators } from 'redux';
import { SelectButton, TextInput } from '../../common';
import InputLabel from '@material-ui/core/InputLabel'
import Button from '@material-ui/core/Button';

class Invite extends Container { 
	constructor(props) {
		super(props);
		this.state = {
			NAME: '',
			EMAIL: '',
			USERTYPE: 0,
			open: false,
			error: ''
		};

		const toController = {
			callback: this.callback,
			getState: this.getState,
			getProps: this.getProps,
		};

		this.controller = new InviteController(toController);
	}

	render() {
		const { handleChange, handleOpen, handleClose, handleSubmit } = this.controller;
		const { NAME, EMAIL, USERTYPE, open } = this.state;
		return (
            <div className='invite'>
			 <InputLabel htmlFor="name-simple">Nome</InputLabel>
				<TextInput
				 id='NAME'
				 value={NAME}
				 changeAction={handleChange}
				/>
				<InputLabel htmlFor="name-simple">Email</InputLabel>
				<TextInput
				 id='EMAIL'
				 value={EMAIL}
				 changeAction={handleChange}
				/>
				<InputLabel htmlFor="demo-controlled-open-select">Tipo de usu√°rio</InputLabel>
				<SelectButton 
					handleOpen={handleOpen}
					handleClose={handleClose}
					handleChange={handleChange}
					userType={USERTYPE}
					dropDownState={open}
				/>
				<div>
				 <Button variant="contained" color="primary" className='button-submit' onClick={handleSubmit}>
      			 Convidar
      			 </Button>	
			    </div>
            </div>
			
		);
	}
} 

const mstp = state => {
	return {};
};


export default connect(mstp, {})(Invite); 