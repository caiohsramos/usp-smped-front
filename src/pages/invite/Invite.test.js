import React from 'react';
import Invite from './Invite';
import renderer from 'react-test-renderer';
import { createStore } from 'redux';
import reducer from '../../reducers';

const store = createStore(reducer);

it('renders correctly', () => {
    const tree = renderer
        .create(<Invite store={store} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});