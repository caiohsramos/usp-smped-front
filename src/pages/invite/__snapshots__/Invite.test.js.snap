// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="invite"
>
  <label
    className="MuiFormLabel-root-12 MuiInputLabel-root-1 MuiInputLabel-animated-9"
    htmlFor="name-simple"
  >
    Usuário
  </label>
  <div
    className="text-input"
    id="text-input-NAME"
  >
    <input
      className="input"
      id="NAME"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <label
    className="MuiFormLabel-root-12 MuiInputLabel-root-1 MuiInputLabel-animated-9"
    htmlFor="name-simple"
  >
    Email
  </label>
  <div
    className="text-input"
    id="text-input-EMAIL"
  >
    <input
      className="input"
      id="EMAIL"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
  <label
    className="MuiFormLabel-root-12 MuiInputLabel-root-1 MuiInputLabel-animated-9"
    htmlFor="demo-controlled-open-select"
  >
    Tipo de usuário
  </label>
  <div>
    <div
      className="MuiFormControl-root-19 select-form"
    >
      <div
        className="MuiInputBase-root-43 MuiInput-root-30 MuiInput-underline-34 MuiInputBase-formControl-44 MuiInput-formControl-31"
        onClick={[Function]}
      >
        <div
          className="MuiSelect-root-23"
        >
          <div
            aria-haspopup="true"
            aria-pressed="false"
            className="MuiSelect-select-24 MuiSelect-selectMenu-27 MuiInputBase-input-53 MuiInput-input-38"
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
          >
            Selecione
          </div>
          <input
            id="demo-controlled-open-select"
            name="userType"
            type="hidden"
            value={0}
          />
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-60 MuiSelect-icon-29"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <label
    className="MuiFormLabel-root-12 MuiInputLabel-root-1 MuiInputLabel-animated-9"
    htmlFor="name-simple"
  >
    Mensagem de convite
  </label>
  <div
    id="text-input-MENSAGEM"
  >
    <textarea
      className="input"
      id="MENSAGEM"
      onChange={[Function]}
      type="text"
      value="Voce foi convidado(a) para SMPED"
    />
  </div>
  <div>
    <button
      className="MuiButtonBase-root-99 MuiButton-root-73 MuiButton-contained-84 MuiButton-containedPrimary-85 MuiButton-raised-87 MuiButton-raisedPrimary-88 button-submit"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="button"
    >
      <span
        className="MuiButton-label-74"
      >
        Convidar
      </span>
      <span
        className="MuiTouchRipple-root-102"
      />
    </button>
  </div>
</div>
`;
